### PR TITLE
Smtp-Email

### PR DESCRIPTION
**Pull request by gherghelalessandro**

### CODE REVIEW

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
+﻿using Assist.July._2022.BE2.Domain;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Assist.Jully._2022.BE2.Controllers
+{
+    [Route("api/[controller]"), ApiController, Authorize]
+    public class usercontroller : ControllerBase
```
Comment (raduflocea) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:8]: PascalCase on class name plz

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                return new OkObjectResult("ok");
+            }
+            catch
+            {
+                return new BadRequestObjectResult("An error has occured");
+            }
+        }
+        [HttpPost("reset/password")]
+        public IActionResult resetpassword()
```
Comment (raduflocea) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:36]: PascalCase on method name

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                return new OkObjectResult("ok");
+            }
+            catch
+            {
+                return new BadRequestObjectResult("An error has occured");
+            }
+        }
+        [HttpGet("{id}")]
+        public IActionResult Getuser(int id)
```
Comment (raduflocea) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:60]: PascalCase on method name

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/appsettings.json [hunk 4]
```diff
@@ -5,5 +5,12 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
-  "AllowedHosts": "*"
+  "AllowedHosts": "*",
+  "MailSettings": {
+    "Mail": "<fromemail>",
```
Comment (raduflocea) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/appsettings.json:10]: You can already create the sender email and fill the Email, Password and DisplayName fields with the according data

File: source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Infrastructure/Services/MailService.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            }
+            
+            builder.HtmlBody = mailRequest.Body;
+            email.Body = builder.ToMessageBody();
+            using var smtp = new SmtpClient();
+            smtp.Connect(_mailSettings.Host, _mailSettings.Port, SecureSocketOptions.StartTls);
+            smtp.Authenticate(_mailSettings.Mail, _mailSettings.Password);
+            await smtp.SendAsync(email);
+            smtp.Disconnect(true);
+
```
Comment (raduflocea) [inline source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Infrastructure/Services/MailService.cs:48]: Unnecessary blank lines

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs:5]: Delete unnecessary usings if not needed.

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Assist.July._2022.BE2.Domain;
+using Assist.July._2022.BE2.Infrastructure;
+[Route("api/[controller]"), ApiController]
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs:9]: one new line.

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+public class MailController : ControllerBase
+{
+    private readonly IMailService mailService;
+    public MailController(IMailService mailService)
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs:13]: one new line.

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        this.mailService = mailService;
+    }
+    [HttpPost("send")]
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs:17]: new line.

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    public async Task<IActionResult> SendMail([FromForm] MailRequest request)
+    {
+        try
+        {
+            await mailService.SendEmailAsync(request);
+            return Ok();
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs:23]: new line, it should be return new OkObjectResult

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        catch (Exception ex)
+        {
+            throw;
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/MailController.cs:27]: it should return a BadRequest

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 12]
```diff
@@ -1,14 +1,12 @@
...
-using Assist.July._2022.BE2.Domain;
+﻿using Assist.July._2022.BE2.Domain;
 using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
 
 namespace Assist.Jully._2022.BE2.Controllers
 {
     [Route("api/[controller]"), ApiController, Authorize]
-    public class usercontroller : ControllerBase
+    public class UserController : ControllerBase
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:8]: new line

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 13]
```diff
@@ -1,14 +1,12 @@
...
     {
-        
         [HttpPost("authenticate")]
         public IActionResult Authenticate(UserAccess Access)
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:11]: new line

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 14]
```diff
@@ -58,7 +56,7 @@ public IActionResult Update(int id)
             }
         }
         [HttpGet("{id}")]
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:58]: new line

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs [hunk 15]
```diff
@@ -34,7 +32,7 @@ public IActionResult Register()
             }
         }
         [HttpPost("reset/password")]
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:34]: new line

File: source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs (hunk 14 above)
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.Jully.2022.BE2/Controllers/UserController.cs:58]: new line

File: source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Domain/Entities/Message.cs [hunk 16]
```diff
@@ -1,9 +1,4 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-namespace Assist.July._2022.BE2.Domain.Entities
+﻿namespace Assist.July._2022.BE2.Domain.Entities
 {
     internal class Message
```
Comment (SebastianFulgaAssist) [inline source/Assist.Jully.2022.BE2/Assist.July._2022.BE2.Domain/Entities/Message.cs:3]: this class should be public

**Review by raduflocea — APPROVED** [review]
Can be merged

**Pull request merged**